packages/react-provider/src/index.tsx

```diff
@@ -49,33 +54,51 @@ export const registerMiddleware = _registerMiddleware;
 
 export type {
   Brand,
+  EventType,
   IActionElemental,
   ICourierContext,
   ICourierEventMessage,
   IInboxMessagePreview,
+  Interceptor,
   ITextElemental,
   Middleware,
+  OnEvent,
   PinDetails,
+  ProviderTheme,
   WSOptions,
-  EventType,
-  OnEvent,
-  Interceptor,
 };
 
+const GlobalThemeVariables = createGlobalStyle<{
+  theme: {
+    variables: ThemeVariables;
+  };
+}>(({ theme }) => {
+  return {
+    ":root": {
+      "--ci-background": theme?.variables?.background,
+      "--ci-text-color": theme?.variables?.textColor,
+      "--ci-title-color": theme?.variables?.titleColor,
+      "--ci-structure": theme?.variables?.structure,
+      "--ci-icon": theme?.variables?.icon,
+    },
+  };
+});
+
 export const CourierContext =
   React.createContext<ICourierContext | undefined>(undefined);
 
 export const CourierProvider: React.FunctionComponent<
   PropsWithChildren<ICourierProviderProps>
 > = ({
-  tenantId,
   apiUrl,
   authorization,
   brand,
   brandId,
   children,
   clientKey,
   id,
+  tenantId,
+  theme: _theme,
```

Comment:
just curious on the rename to `_theme`?

Reply:
because i merge it with some defaults later and id rather call that "theme" than "mergedTheme" i guess

Reply:
and i do this pattern already with _middleware